src/slices/filters.ts

```diff
@@ -47,6 +57,38 @@ const reducers = {
       });
     });
   },
+  toggleFilterOption: (state: FiltersState, { payload }: PayloadAction<FilterPayload>) => {
+    if (!state.static) {
+      state.static = {};
+    }
+    const { filterCollectionId, filter: targetFilter, shouldSelect } = payload;
+    if (!state.static[filterCollectionId]) {
+      if (filterCollectionId && shouldSelect) {
+        state.static[filterCollectionId] = [];
+      } else {
+        console.warn(`invalid static filters id: ${filterCollectionId}`);
```

Comment:
Could we simplify this a bit by moving the filterCollectionId check near the top of the function? That way we could have : `if (!filterCollectionId) { console.warn('invalid...') }` and then the logic here could remove the double if statement and instead have `if (!state.static[filterCollectionId] && shouldSelect) { ...`. We may not need the `shouldSelect` in the if statement either. If the filter isn't in the state and shouldSelect is false, I think it's okay to put that in the state so long as it's marked as not selected

Reply:
if a user somehow unselect something that is not part of the state (never selected), that is an odd behavior no?

Reply:
ah yeah, will do